src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -309,31 +399,45 @@ public void CopyChangedValues(TStructuralType original)
                 // Patch for each nested resource changed under this TStructuralType.
                 dynamic deltaNestedResource = _deltaNestedResources[nestedResourceName];
                 dynamic originalNestedResource = null;
-                if (!TryGetPropertyRef(original, nestedResourceName, out originalNestedResource))
-                {
-                    throw Error.Argument(nestedResourceName, SRResources.DeltaNestedResourceNameNotFound,
-                        nestedResourceName, original.GetType());
-                }
 
-                if (originalNestedResource == null)
+                if(deltaNestedResource is IDeltaSet)
                 {
-                    // When patching original target of null value, directly set nested resource.
-                    dynamic deltaObject = _deltaNestedResources[nestedResourceName];
-                    dynamic instance = deltaObject.GetInstance();
-
-                    // Recursively patch up the instance with the nested resources.
-                    deltaObject.CopyChangedValues(instance);
+                    IPatchMethodHandler patchHandler = PatchHandler.GetNestedPatchHandler(original, nestedResourceName);
 
-                    _allProperties[nestedResourceName].SetValue(original, instance);
+                    if (patchHandler != null)
```

Comment:
>if (patchHandler != null) [](start = 20, length = 25)

what if patchHandler == null?  should this silently fail?

Reply:
This would mean they dont want (intentionally or unintentionally) to patch that nested item, so it would be skpped, right>?